packages/devextreme/js/__internal/ui/file_uploader/file_uploader.ts

```diff
@@ -1399,17 +1426,38 @@ class FileUploader extends Editor<FileUploaderProperties> {
   }
 
   _renderFileButtons(file: FileUploaderItem, $container: dxElementWrapper): void {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { _cancelButtonPosition } = this.option();
+
+    const $uploadButton = this._getUploadButton(file);
+
+    if ($uploadButton) {
+      $container.prepend($uploadButton);
+    }
+
     const $cancelButton = this._getCancelButton(file);
 
     if ($cancelButton) {
-      $container.append($cancelButton);
+      if (_cancelButtonPosition === 'end') {
+        $container.append($cancelButton);
+
+        return;
+      }
+      $container.prepend($cancelButton);
     }
+  }
 
-    const $uploadButton = this._getUploadButton(file);
+  _renderFileIcon(fileName: string, $container: dxElementWrapper): void {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { _showFileIcon } = this.option();
 
-    if ($uploadButton) {
-      $container.append($uploadButton);
+    if (!_showFileIcon) {
+      return;
     }
+
+    $('<div>')
+      .addClass(`${FILEUPLOADER_FILE_ICON_CLASS} dx-icon dx-icon-${getFileIconName(fileName)}`)
```

Comment:
Let's move `dx-icon` to a constant too 
```suggestion
      .addClass(`${FILEUPLOADER_FILE_ICON_CLASS} ${ICON_CLASS} ${ICON_CLASS}-${getFileIconName(fileName)}`)
```